client/gutenberg/extensions/slideshow/slideshow.js

```diff
@@ -49,26 +49,41 @@ class Slideshow extends Component {
 
 		return (
 			<div className={ className } data-effect={ effect }>
-				<div className="swiper-container" ref={ this.slideshowRef }>
+				<div
+					className="wp-block-jetpack-slideshow_container swiper-container"
```

Comment:
Should we drop the swiper specific classes? This would allow us to keep the markup more vendor agnostic.

It seems like that all of the container, wrapper, slide, pagination, and navigation selectors can be set to custom classes (or HTML elements, in some cases) at creation time ([#](https://idangero.us/swiper/api/)).

Reply:
I believe the swiper-specific classes are important and come with a good deal of necessary CSS. I did a few experiments removing them and setting some of Swiper's CSS class parameters e.g. (`slideClass="wp-block-jetpack-slideshow_slide"`). This didn't go well—much of Swiper's style and functionality broke—which lead me to believe you can only use those class reassignment parameters if you plan to duplicate/replace a large amount of Swiper's CSS, which we probably don't want to do. That said, I didn't dig terribly deeply. As soon as I sensed this was not going to be a simple task I let it drop.

Reply:
Uh okay, yeah that would be a dealbreaker if Swiper's styling ceased to work...

Reply:
I also can't get this to work. Seems like [styling is really hard-wired](https://github.com/nolimits4web/swiper/blob/cf9268762cde0b502ab192ce28f038726371e622/src/components/core/core.less).

For the record, this is what I tried:

```diff
diff --git a/client/gutenberg/extensions/slideshow/create-swiper.js b/client/gutenberg/extensions/slideshow/create-swiper.js
index 13bc799077..2994b41cd8 100644
--- a/client/gutenberg/extensions/slideshow/create-swiper.js
+++ b/client/gutenberg/extensions/slideshow/create-swiper.js
@@ -5,7 +5,10 @@ import { merge } from 'lodash';
 
 const SIXTEEN_BY_NINE = 16 / 9;
 
-export default async function createSwiper( container = '.swiper-container', params = {} ) {
+export default async function createSwiper(
+	container = '.wp-block-jetpack-slideshow_container',
+	params = {}
+) {
 	const autoSize = function() {
 		const img = this.slides[ 0 ].querySelector( 'img ' );
 		const aspectRatio = img.clientWidth / img.clientHeight;
@@ -21,18 +24,20 @@ export default async function createSwiper( container = '.swiper-container', par
 		init: true,
 		initialSlide: 0,
 		navigation: {
-			nextEl: '.swiper-button-next',
-			prevEl: '.swiper-button-prev',
+			nextEl: '.wp-block-jetpack-slideshow_button-next',
+			prevEl: '.wp-block-jetpack-slideshow_button-prev',
 		},
 		pagination: {
 			clickable: true,
-			el: '.swiper-pagination',
+			el: '.wp-block-jetpack-slideshow_pagination',
 			type: 'bullets',
 		},
 		preventClicksPropagation: false /* Necessary for normal block interactions */,
 		releaseFormElements: false,
 		setWrapperSize: true,
+		slideClass: 'wp-block-jetpack-slideshow_slide',
 		touchStartPreventDefault: false,
+		wrapperClass: 'wp-block-jetpack-slideshow_wrapper',
 		/* We probably won't end up needing both init and imagesReady. Just casting a wide net for now. */
 		on: {
 			init: autoSize,
diff --git a/client/gutenberg/extensions/slideshow/slideshow.js b/client/gutenberg/extensions/slideshow/slideshow.js
index 72e1e76078..ecdede59ae 100644
--- a/client/gutenberg/extensions/slideshow/slideshow.js
+++ b/client/gutenberg/extensions/slideshow/slideshow.js
@@ -49,13 +49,10 @@ class Slideshow extends Component {
 
 		return (
 			<div className={ className } data-effect={ effect }>
-				<div
-					className="wp-block-jetpack-slideshow_container swiper-container"
-					ref={ this.slideshowRef }
-				>
-					<div className="swiper-wrapper">
+				<div className="wp-block-jetpack-slideshow_container" ref={ this.slideshowRef }>
+					<div className="wp-block-jetpack-slideshow_wrapper">
 						{ images.map( ( { alt, caption, id, url } ) => (
-							<figure className="wp-block-jetpack-slideshow_slide swiper-slide" key={ id }>
+							<figure className="wp-block-jetpack-slideshow_slide" key={ id }>
 								<div className="wp-block-jetpack-slideshow_image-container">
 									<img
 										alt={ alt }
@@ -73,15 +70,15 @@ class Slideshow extends Component {
 						) ) }
 					</div>
 					<div
-						className="wp-block-jetpack-slideshow_pagination swiper-pagination swiper-pagination-white"
+						className="wp-block-jetpack-slideshow_pagination swiper-pagination-white"
 						ref={ this.paginationRef }
 					/>
 					<div
-						className="wp-block-jetpack-slideshow_button-prev swiper-button-prev swiper-button-white"
+						className="wp-block-jetpack-slideshow_button-prev swiper-button-white"
 						ref={ this.btnPrevRef }
 					/>
 					<div
-						className="wp-block-jetpack-slideshow_button-next swiper-button-next swiper-button-white"
+						className="wp-block-jetpack-slideshow_button-next swiper-button-white"
 						ref={ this.btnNextRef }
 					/>
 				</div>
diff --git a/client/gutenberg/extensions/slideshow/view.js b/client/gutenberg/extensions/slideshow/view.js
index 2eebeb18c3..384c7123a5 100644
--- a/client/gutenberg/extensions/slideshow/view.js
+++ b/client/gutenberg/extensions/slideshow/view.js
@@ -16,7 +16,9 @@ typeof window !== 'undefined' &&
 		const slideshowBlocks = document.getElementsByClassName( 'wp-block-jetpack-slideshow' );
 		forEach( slideshowBlocks, slideshowBlock => {
 			const { effect } = slideshowBlock.dataset;
-			const slideshowContainer = slideshowBlock.getElementsByClassName( 'swiper-container' )[ 0 ];
+			const slideshowContainer = slideshowBlock.getElementsByClassName(
+				'wp-block-jetpack-slideshow_container'
+			)[ 0 ];
 
 			createSwiper( slideshowContainer, { effect, init: true, initialSlide: 0 } );
 		} );
```

Reply:
Yeah, this is precisely what I tried. I guess we should just stick with Swiper classes.

Reply:
Should we use nested imports to wrap Slider's CSS with our CSS selector?
```scss
.wp-block-jetpack-slideshow {
  @import '......./slider.css';
}
```

http://sass-lang.com/documentation/file.SASS_REFERENCE.html#nested_import

Wouldn't that mostly solve the problems around our CSS leaking into any other implementation of Slider on the same page?

As the term "slider" is pretty generic already and Gutenberg has great deprecation transforms in case we want to change it later, I wouldn't be so worried about having the term in the block output per se.

Reply:
It's `swiper` not `slider`, but your point still stands :-). Currently we're loading Swiper's stylesheet dynamically, so wrapping it isn't possible: https://github.com/Automattic/wp-calypso/blob/713a6a5e7cc27f4467af5bd97b191d5f1dbbcbbd/client/gutenberg/extensions/slideshow/create-swiper.js#L47. However, the main win in the dynamic loading is the large Javascript library, so we could include the stylesheet as you're suggesting without too much pain. I guess this becomes a trade-off between CSS leakage and lazy loading.

Reply:
Let's defer this to unblock this PR -- it seems like this might need some more discussion/experimentation. I'll file an issue.

Reply:
#30630